File: CleanCity/app/src/main/java/com/example/cleancity/modelos/CamionModelo.java
package com.example.cleancity.modelos;

import com.example.cleancity.api.Parametro;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

import kotlin.collections.ArrayDeque;

public class CamionModelo {
    private String matricula;
    private String conductor;
    private double lat;
    private double lng;

    public CamionModelo() {
    }

    public CamionModelo(String matricula, String conductor, double lat, double lng) {
        this.matricula = matricula;
        this.conductor = conductor;
        this.lat = lat;
        this.lng = lng;
    }

    public CamionModelo(JSONObject fromJSON) throws JSONException {
        this.matricula = fromJSON.getString("Matricula_Camion");
        this.conductor = fromJSON.getString("Conductor_Camion");
        this.lat = fromJSON.getDouble("Latitud_Camion");
        this.lng = fromJSON.getDouble("Longitud_Camion");
    }

    public List<Parametro> busqueda(){
        List<Parametro> retornar = new ArrayDeque<Parametro>();
        try{
            retornar.add(new Parametro("Matricula_Camion", getMatricula()));
        }catch (Exception e){}

        return retornar;
    }

    public List<Parametro> parametros(){
        List<Parametro> retornar = new ArrayDeque<Parametro>();
        try{
            retornar.add(new Parametro("Matricula_Camion", getMatricula()));
        }catch (Exception e){}
        retornar.add(new Parametro("Conductor_Camion", getConductor()));
        retornar.add(new Parametro("Latitud_Camion", getLat()));
        retornar.add(new Parametro("Longitud_Camion", getLng()));

        return retornar;
    }

    @Override
    public String toString() {
        return "CamionModelo{" +
                "matricula='" + matricula + '\'' +
                ", nombre='" + conductor + '\'' +
                ", lat=" + lat +
                ", lng=" + lng +
                '}';
    }

    public String getMatricula() {
        return matricula;
    }

    public void setMatricula(String matricula) {
        this.matricula = matricula;
    }

    public String getConductor() {
        return conductor;
    }

    public void setConductor(String conductor) {
        this.conductor = conductor;
    }

    public double getLat() {
        return lat;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public double getLng() {
        return lng;
    }

    public void setLng(double lng) {
        this.lng = lng;
    }
}

File: CleanCityClient/app/src/main/java/com/example/cleancityclient/modelos/CamionModelo.kt
package com.example.cleancityclient.modelos

import com.example.cleancityclient.api.Parametro
import org.json.JSONException
import org.json.JSONObject

class CamionModelo() {
    private var matricula: String? = null
    private var conductor: String? = null
    private var latitud: Double? = null
    private var longitud: Double? = null

    constructor(matricula: String?, conductor: String?, latitud: Double?, longitud: Double?): this(){
        this.matricula = matricula
        this.conductor = conductor
        this.latitud = latitud
        this.longitud = longitud
    }

    @Throws(JSONException::class)
    constructor(fromJSON: JSONObject) : this(){
        matricula = fromJSON.getString("Matricula_Camion")
        conductor = fromJSON.getString("Conductor_Camion")
        latitud = fromJSON.getDouble("Latitud_Camion")
        longitud = fromJSON.getDouble("Longitud_Camion")
    }

    fun busqueda(): List<Parametro>? {
        val retornar: MutableList<Parametro> = ArrayDeque<Parametro>()
        try {
            retornar.add(Parametro("Matricula_Camion", getMatricula()))
        } catch (e: Exception) {
        }
        return retornar
    }

    fun parametros(): List<Parametro>? {
        val retornar: MutableList<Parametro> = ArrayDeque<Parametro>()
        try {
            retornar.add(Parametro("Matricula_Camion", getMatricula()))
        } catch (e: Exception) {
        }
        retornar.add(Parametro("Conductor_Camion", getConductor()))
        retornar.add(Parametro("Latitud_Camion", getLatitud()))
        retornar.add(Parametro("Longitud_Camion", getLongitud()))
        return retornar
    }

    override fun toString(): String {
        return "CamionModelo(matricula=$matricula, conductor=$conductor, latitud=$latitud, longitud=$longitud)"
    }

    fun getMatricula(): String?{ return matricula }

    fun setMatricula(matricula: String?){ this.matricula = matricula }

    fun getConductor(): String?{ return conductor }

    fun setConductor(conductor: String?){ this.conductor = conductor}

    fun getLatitud(): Double? { return latitud}

    fun setLatitud(latitud: Double?){ this.latitud = latitud}

    fun getLongitud(): Double?{ return longitud}

    fun setLongitud(longitud: Double?){ this.longitud = longitud}
}